plugins/nodes/src/js/components/modals/DeactivateNodeConfirm.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import { i18nMark, withI18n } from "@lingui/react";
+
+// @ts-ignore
+import { request } from "@dcos/mesos-client";
+import { Trans } from "@lingui/macro";
+import { InfoBoxInline } from "@dcos/ui-kit";
+
+// @ts-ignore
+import { Confirm } from "reactjs-components";
+
+import Node from "#SRC/js/structs/Node";
+import NodeMaintenanceActions from "#PLUGINS/nodes/src/js/actions/NodeMaintenanceActions";
+
+interface Props {
+  open: boolean;
+  onClose: () => void;
+  node: Node | null;
+  i18n: any;
+}
+
+function DeactivateNodeConfirm(props: Props) {
+  const { open, onClose, node, i18n } = props;
+
+  const [inProgress, setInProgress] = useState<boolean>(false);
+  const [networkError, setNetworkError] = useState<React.ReactElement | null>(
+    null
+  );
+
+  let [prevOpenState, setPrevOpenState] = useState<boolean | null>(null);
+
+  // Reset form when closed
+  if (open !== prevOpenState) {
```

Comment:
I wonder if there could be a handler to extend `props.onClose` to perform these state changes, to avoid having to keep track of `prevOpenState`?

Reply:
Good idea